tests/gitlab/build_and_test

```diff
@@ -230,13 +232,21 @@ then
     echo "~~~~~ Testing MFEM"
     echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
 
-    return_code=0
-    make test 2>&1 | tee tests_output.txt || return_code=${?}
+    make test
+    test_status=$?
 
-    if grep -q "All tests passed." ./tests_output.txt
-    then
-        echo "SUCCESS: No error during tests" && exit 0
+    echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+    echo "~~~~~ Cleaning MFEM"
+    echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+
+    if make distclean > distclean.log 2>&1; then
```

Comment:
@v-dobrev Have you seen the MFEM files growing _indefinitely_?

I checked and if I am not mistaken, the build happens in the CI_PROJECT_DIR (a clone of the MFEM repo). 
However, this directory is reused by subsequent pipelines in a random manner:

Gitlab creates one of those clones per runner x the number of jobs handled by a runner (typically 8). This means that the number of MFEM clones in the file system is finite and they are recycled.

That being said, it is not wrong to clean the build directory, I am just thinking that this will not solve a possible "storage leak".

Reply:
I don't think there is a "leak" that causes indefinite growth in the disk space used by gitlab. Cleaning the MFEM builds helps reduce the disk usage quite significantly -- each build can be several gigabytes, so things add up quickly.